amm-executor/src/Spectrum/Executor/OrdersExecutor/Service.hs

```diff
@@ -0,0 +1,116 @@
+  logging         <- forComponent "OrdersExecutor"
+  pure $ OrdersExecutor
+    { run = run' logging backlog transactions resolver poolActions
+    }
+
+run'
+  :: forall s m era. (IsStream s, MonadAsync m, MonadUnliftIO m)
+  => Logging m
+  -> Backlog m
+  -> Transactions m era
+  -> PoolResolver m
+  -> PoolActions
+  -> s m ()
+run' logging@Logging{..} backlog@Backlog{..} txs resolver poolActions =
+  S.repeatM tryAcquire & S.mapM (\case
+      Just order ->
+        infoM ("Going to execute order for pool" ++ show order) >> execute' logging backlog txs resolver poolActions order
+      Nothing    ->
+        pure ()
+    )
+
+execute' :: forall m era. (MonadUnliftIO m, MonadThrow m) => Logging m -> Backlog m -> Transactions m era -> PoolResolver m -> PoolActions -> Order -> m ()
+execute' logging backlog@Backlog{suspend, drop} txs resolver poolActions order =
+  catch (executeOrder' backlog txs resolver poolActions order) (\(err :: OrderExecErr) -> case err of
+      PriceTooHigh  -> attachLogging logging PriceTooHigh order suspend $ SuspendedOrder order
+      dropErr       -> attachLogging logging dropErr order drop $ orderId order
+    )
+
+executeOrder' :: (Monad m, MonadThrow m) => Backlog m -> Transactions m era -> PoolResolver m -> PoolActions -> Order -> m ()
+executeOrder' Backlog{checkLater} Transactions{..} PoolResolver{..} poolActions order@(OnChain _ Core.AnyOrder{..}) = do
+  mPool <- resolvePool anyOrderPoolId
+  pool@(OnChain prevPoolOut Core.Pool{poolId})  <- throwMaybe (EmptyPool anyOrderPoolId) mPool
+  (txCandidate, Predicted _ predictedPool)      <- throwEither $ runOrder pool order poolActions
+  tx    <- finalizeTx txCandidate
+  pPool <- throwMaybe (PoolNotFoundInFinalTx poolId) (extractPoolTxOut pool tx)
+  let
+    tracedPredictedPool = Traced
+      { tracedState  = State.Predicted (OnChain pPool predictedPool)
+      , prevTxOutRef = fullTxOutRef prevPoolOut
+      }
+  submitTx tx
+  putPool tracedPredictedPool
+  checkLater (InProgressOrder order)
+  pure ()
+
+extractPoolTxOut :: forall era. Pool -> Tx era -> Maybe FullTxOut
+extractPoolTxOut (OnChain poolOutput _) tx = List.find (\output -> fullTxOutDatum output == fullTxOutDatum poolOutput) (Interop.extractCardanoTxOutputs tx)
+
+runOrder
+  :: Pool
+  -> Order
+  -> PoolActions
+  -> Either OrderExecErr (TxCandidate, Predicted Core.Pool)
+runOrder (OnChain poolOut pool) (OnChain orderOut Core.AnyOrder{..}) PoolActions{..} =
+  case anyOrderAction of
+    DepositAction deposit -> runDeposit (Confirmed orderOut deposit) (poolOut, pool)
+    RedeemAction redeem   -> runRedeem (Confirmed orderOut redeem) (poolOut, pool)
+    SwapAction swap       -> runSwap (Confirmed orderOut swap) (poolOut, pool)
+
+attachLogging :: Monad m => Logging m -> OrderExecErr -> Order -> (a -> m Bool) -> (a -> m ())
+attachLogging Logging{..} err order ma input = ma input >>= (\opRes -> infoM (show err ++ " for " ++ show order ++ ". Op result:" ++ show opRes))
```

Comment:
`logResult` is better. At first I was confused by attachLogging as it is used to attach logging algebra to a module.